app/controllers/courses_controller.rb

```diff
@@ -58,7 +58,7 @@ def statistics
       if current_user.id == @teacher_id
         @teacher = true
       elsif current_user.student?
-        @student = @terms.first.students.find_by user_id: current_user.id
+        @student = @terms.collect { |term| term.students.where(user_id: current_user.id) }.any?
```

Comment:
I've managed to do it like
```ruby
@current_user.subscriptions.where(term_id: @course.terms.pluck(:id)).any?
```
It can be much faster on the actual database, but I haven't checked it yet. `pluck` should be efficient here.
I'll benchmark it later.